romancal/flux/flux_step.py

```diff
@@ -104,26 +99,19 @@ def apply_flux_correction(model):
     DATA = ("data", "err")
     VARIANCES = ("var_rnoise", "var_poisson", "var_flat")
 
-    if model.data.unit == model.meta.photometry.conversion_megajanskys.unit:
-        log.info(
-            f"Input data is already in flux units of {model.meta.photometry.conversion_megajanskys.unit}."
-        )
-        log.info("Flux correction already applied.")
-        return
-
-    if model.data.unit != LV2_UNITS:
+    if model.meta.cal_step["flux"] == "COMPLETE":
         message = (
-            f"Input data units {model.data.unit} are not in the expected units of {LV2_UNITS}"
-            "\nAborting flux correction"
+            "Input data is already in flux units of MJy/sr."
+            "\nFlux correction already applied."
         )
-        [log.error(line) for line in message.splitlines()]
-        raise ValueError(message)
+        log.info(message)
+        return
```

Comment:
I think we should just delete this stanza.  Do we special case the "already complete" step anywhere else?